libs/commons-database/src/main/kotlin/gov/cdc/ocio/database/dynamo/DynamoCollection.kt

```diff
@@ -0,0 +1,126 @@
+package gov.cdc.ocio.database.dynamo
+
+import mu.KotlinLogging
+import gov.cdc.ocio.database.persistence.Collection
+import software.amazon.awssdk.enhanced.dynamodb.*
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient
+import software.amazon.awssdk.services.dynamodb.model.ExecuteStatementRequest
+
+
+/**
+ * DynamoDB collection implementation
+ *
+ * @property enhancedClient DynamoDbEnhancedClient
+ * @property dynamoTableName String
+ * @property tableClassType Class<R>
+ * @property dynamoTable DynamoDbTable<R>?
+ * @constructor
+ */
+class DynamoCollection<R>(
+    private val dynamoDbClient: DynamoDbClient,
+    private val enhancedClient: DynamoDbEnhancedClient,
+    private val dynamoTableName: String,
+    private val tableClassType: Class<R>
+) : Collection {
+
+    private val logger = KotlinLogging.logger {}
+
+    private val dynamoTable = collectionForClass(tableClassType)
+
+    /**
+     * Builds a dynamodb table schema from the provided class type.
+     *
+     * @param classType Class<R>
+     * @return DynamoDbTable<R>?
+     */
+    private fun <R> collectionForClass(classType: Class<R>): DynamoDbTable<R>? {
+        return try {
+            enhancedClient.table(
+                dynamoTableName,
+                TableSchema.fromClass(classType)
+            )
+        } catch (e: Exception) {
+            logger.error("Unable to establish the dynamodb enhanced client for table: ${classType.name} with exception: $e")
+            null
```

Comment:
Since this is used for property assignment in the class I'd prefer this return null and not throw an exception since the property creation will not catch it and the exception would get propagated up to the creator of this class, which is not what we want.